include/pybind11/subinterpreter.h

```diff
@@ -117,6 +132,7 @@ class subinterpreter {
         // same as the default config in the python docs
         PyInterpreterConfig cfg;
         std::memset(&cfg, 0, sizeof(cfg));
+        cfg.allow_threads = 1;
```

Comment:
There is this existing comment above:
```
// same as the default config in the python docs
```
Does that need to be adjusted, to be truthful?

We seem to have a problem only with Python 3.12. Did you consider something like this:
```
#if PY_VERSION_HEX < 0x030D0000
cfg.allow_threads = 1;
#endif
cfg.allow_threads = 0;
#endif
```

I totally don't know if that makes sense, but it seems important to keep the comment up-to-date with the rationale for the config choices here.

Or just remove the comment, if it's too much trouble explaining the choices here.